### ISSUE TITLE
Add simple values() method in Tensorics to retrieve all the values of a Tensor

### ISSUE THREAD

**Issue by delph74** [tensorics/tensorics-core#42]

**Comment by kaifox** [comment]
Hi @delph74 

What would you expect as a return type of such a method?

**Comment by delph74** [comment]
Hello @kaifox ,
We could return a Set<typeOfTheTensor>, and in this case maybe call the method uniqueValues().
What do you think?

**Comment by agorzawski** [comment]
Hi @delph74  and @kaifox 

As far I remember, a tensor has no 'values' like method. But (if implements Mappable, and this is the case for the ImmutableTensor) it has `.asMap()`... which you follow with `values()`.

The practical use of such values collection (only values) may be very limited, as this may vary with their position in the Tensor (coordinate).

**Comment by kaifox** [comment]
yes, then its clear!

Potentially  uniqueValuesOf(...) as the other methods in Tensorics?
```java
public static final <T> Set<T> Tensorics.uniqueValuesOf(Tensor<T> tensor);
```

In general, I clients should void using also the asMap() method directly ... In general, Tensorics.mapFrom(Tensor t) should be used. This calls asMap() if available, which makes it more efficient....

**Comment by kaifox** [comment]
Hey @agorzawski !!! Cool to see you here!

**Comment by agorzawski** [comment]
@kaifox I just want to recall, that I have never left this project ;-)

**Comment by delph74** [comment]
Yes @kaifox , something like that would be perfect. 
I often use values for Map and I think for a Tensor we may have many usecase where such method would be needed!

**Comment by michi42** [comment]
Maybe one more thing - there is `Tensorics.stream(Tensor)` that allows processing Position-Value pairs via the Java 8 streaming API, and `TensorStreamFilters` and `TensorStreamMappers` to help with filtering/mapping. Also there are some collectors to build tensors or tensorbackeds with streams in `TensorStreams`.

**Comment by delph74** [comment]
yes of course it it possible and not to difficult to get all the values of a Tensor, like that for example
Tensorics.stream(betaStarParameters).map(Map.Entry::getValue).collect(toSet());

So nothing urgent or really missing, just for convenience and make it easier.

**Comment by kaifox** [comment]
Good point @michi42 ! ... Still, convenience methods can still be justified... I think there is nothing against adding it.

**Comment by kaifox** [comment]
@agorzawski , good to hear!

**Comment by michi42** [comment]
Yes sure, I did not mean to object that :-)

**Comment by kaifox** [comment]
I know ;-)

**Comment by kaifox** [comment]
Just wanted to encourage @delph74  to add it if she feels like ;-)

**Comment by delph74** [comment]
Yes why not, I would then add my first contribution to tensoric project :v:  😃

**Comment by kaifox** [comment]
Cooool!!!!!
